python/lsst/analysis/drp/histPlot.py

```diff
@@ -274,94 +236,95 @@ def histPlot(self, catPlot, plotInfo, sumStats):
-        leftPanelAx.set_xlabel(self.config.axisLabels["xLeft"])
-        leftPanelAx.set_ylabel(self.config.axisLabels["yLeft"])
-        leftPanelAx.tick_params(labelsize=7)
-        leftPanelAx.legend(loc="upper left", bbox_to_anchor=(0.00, 0.77),
-                           bbox_transform=fig.transFigure, fontsize=7, handleheight=1.5)
-        leftPanelAx.axvline(0, ls='--', lw=2, c='k')
-
-        # right panel limits
-        pLowers, pUppers = [], []
-        for key in self.config.rightPanelActions.fieldNames:
-            column = self.config.rightPanelLabels[key]
-            [pLower, pUpper] = np.nanpercentile(catPlot[column].values,
-                                                [self.config.pLower, self.config.pUpper])
-            pLowers.append(pLower)
-            pUppers.append(pUpper)
-        rightPanelLower = np.min(pLowers)
-        rightPanelUpper = np.max(pUppers)
-
-        # right panel plotting
-        rightPanelAx = fig.add_subplot(gs[:, 66:])
-        plt.rcParams["hatch.color"] = "white"
-        for count, key in enumerate(self.config.rightPanelActions.fieldNames):
-            column = self.config.rightPanelLabels[key]
-            mask = np.isfinite(catPlot[column].values)
-            label = f"{column} ({np.sum(mask)})"
-            col = plt.cm.tab10(count)
-            rightPanelAx.hist(catPlot[column][mask], bins=self.config.nBins, alpha=0.7, color=col,
-                              range=(rightPanelLower, rightPanelUpper), histtype="bar", lw=2, label=label,
-                              hatch="//")
-            datMed = np.nanmedian(catPlot[column][mask])
-            datMad = sigmaMad(catPlot[column][mask])
-            rightPanelAx.text(x=0.03, y=0.985-(count*0.08), transform=rightPanelAx.transAxes,
-                              ha="left", va="top", fontsize=7, c=col,
-                              s=f"med = {datMed:0.2f}\n$σ_{{MAD}}$ = {datMad:0.2f}")
-            rightPanelAx.axvline(datMed, ls=':', lw=2, c=col)
-        rightPanelAx.set_xlim(rightPanelLower, rightPanelUpper)
-        rightPanelAx.set_xlabel(self.config.axisLabels["xRight"])
-        rightPanelAx.tick_params(labelsize=7)
-        yStep = 0.038*len(self.config.leftPanelActions.fieldNames) + 0.01
-        rightPanelAx.legend(loc="upper left", bbox_to_anchor=(0.00, 0.77-yStep),
-                            bbox_transform=fig.transFigure, fontsize=7, handleheight=1.5)
-        rightPanelAx.axvline(0, ls='--', lw=2, c='k')
-
-        # Corner plot of patches showing summary stat in each
-        axCorner = plt.gcf().add_subplot(gs[70:96, 3:21])
+        gs = gridspec.GridSpec(240, 240)
+
+        # determine gridspec divisions
+        summary_block = 40
+        panel_pad = 8
+        if len(panels) <= 2:
+            ncols = 1
+        else:
+            ncols = 2
+        nrows = int(np.ceil(len(panels) / ncols))
+        col_bounds = np.linspace(0+summary_block+panel_pad, 240, ncols+1, dtype=int)
```

Comment:
Spaces needed.

Reply:
Here and throughout, did you run PEP8?

Reply:
The entire `histPlot.py` script has been PEP8'ed. With the exception of `E501 line too long`, all PEP8 violations have been resolved.